### PR TITLE
Improve performance of RAW FRAMED mode

### PR DESCRIPTION
**Pull request by thomascobb**

### CODE REVIEW

**Comment by thomascobb** [comment]
Performance for 45MB/s:
- Baseline: 50%
- 4MB buffer: 44%
- 4MB buffer + passthrough: 33%

File: server/data_server.c [hunk 1]
```diff
@@ -434,18 +434,20 @@ static bool write_block_base64(
 }
 
 
+static void set_frame_header(void *buffer, uint32_t frame_length) {
+    /* Update the first four bytes of the buffer with a header followed by a
+        * frame byte count. */
```
Comment (Araneidae) [inline server/data_server.c:439]: Is this a tab I see before me?  If not, this is a funny looking indentation error.

Please check for both errant tabs and trailing whitespace, which I don't seem to be able to see in this view.

Comment (Araneidae) [reply server/data_server.c:439]: Have just added a plugin (Render Whitespace on GitHub, really hope it's safe), and this appears to just be dodgy layout!

Comment (thomascobb) [reply server/data_server.c]: yup, too many spaces...

File: server/capture.c [hunk 2]
```diff
@@ -425,6 +426,7 @@ static void ensure_sample_count(
 
     /* Not being captured.  We need to emit a capture entry right now.  Because
      * we're first and only, we go into the anonymous group. */
+    capture->sample_count_anonymous = true;
```
Comment (Araneidae) [inline server/capture.c:429]: You're setting this here ... but where is it being reset?  Don't think it is...

Comment (thomascobb) [reply server/capture.c:429]: I'm relying on it being initialized to false when the structure is created. Does that work?

Comment (thomascobb) [reply server/capture.c:429]: ahh, I see it persists...

File: server/capture.c [hunk 3]
```diff
@@ -448,6 +450,7 @@ static void gather_data_capture(
 {
     /* Work through the fields. */
     struct data_capture *capture = gather->capture;
+    capture->sample_count_anonymous = false;
```
Comment (Araneidae) [inline server/capture.c:453]: I'm not happy about putting this here.  The problem is that you're assigning to `sample_count_anonymous` in two different places, and this line here isn't a full initialisation of `*capture`.  You could perhaps do
```
*capture = (struct data_capture) {};
```
here instead, but I'd want to know that this was the right time and place for this.

Anyway, hopefully we'll talk about this and related issues later today.

File: server/ext_out.c [hunk 4]
```diff
@@ -41,6 +41,7 @@ struct ext_out {
 
 
 static struct ext_out *samples_ext_out;
+static char samples_field_name[MAX_NAME_LENGTH];
```
Comment (Araneidae) [inline server/ext_out.c:44]: I'm confused: don't we already have a mechanism for giving the samples a field name?  After all, sample count is a legitimate selection, thought we were faking that?

**Comment by Araneidae** [comment]
Impressive results.

My only caveat is the change in behaviour to `RAW FRAMED`, but I don't think it's worth the hassle of creating another mode.

Oh, @thomascobb , while we're at it, we should document now that the frame size can be up to 2MB.  Don't think the frame size has been documented, and it was previously quite a bit smaller (64K?).  Am happy to merge this as it stands.

**Pull request merged**